### ISSUE TITLE
[3] Enable searching for labels in catalog controller.

### ISSUE THREAD

**Issue by straleyb** [OregonDigital/OD2#272]
### Descriptive summary

There will be some extra work involved to configure the system to search the controlled fields properly. Verify that all the searching works for each field. that means searching throughout the system, review page, search bar, facets, etc.

**Comment by CGillen** [comment]
@wickr Pointed out this still needs configuration for `config.add_search_field`

**Comment by CGillen** [comment]
QA:
In all expected search fields/pages:
 - [ ] Can still search on core metadata
 - [ ] Can search on some obscure custom metadata
 - [ ] Searches work for labels for controlled field metadata and not the URI
 - [ ] Cannot search on metadata marked not searchable by MAP

**Comment by vantuyls** [comment]
QA blocked by #463

**Comment by vantuyls** [comment]
QA Fail - MAP indicated field Specimin Type should be non-searchable but it is.

Please check all searchable, facetable, etc. fields of MAP and compare with current configuration.

**Comment by wickr** [comment]
QA Result (joint work of Metadata Team, also marked in MAP, first column)
- [x] Titles - pass
- [x] Creators - pass
- [ ] Descriptions - fail
  - [ ] Abstract - can't search
  - [ ] Compass Direction - missing from showpage
  - [ ] Description - missing from edit form
  - [ ] Form of Work - missing from showpage, no facet, can't search
  - [ ] Identification Verification Status - missing from showpage
  - [ ] Layout - missing from showpage
  - [ ] Mode of Issuance - missing from showpage
  - [ ] MODS Note - missing from showpage
  - [ ] Number of Pages - missing from showpage
  - [ ] Object Orientation - missing from showpage
  - [ ] Tribal Notes - missing from showpage
  - [ ] Source Condition - missing from showpage
  - [ ] Specimen Type - missing from showpage
  - [ ] Temporal - missing from showpage
- [x] Subjects - pass
- [ ] Geographics - fail
  - [ ] Box - missing from showpage
  - [ ] GPS Latitude - missing from showpage
  - [ ] GPS Longitude - missing from showpage
  - [ ] Location - with Document type, missing from showpage, not searchable, not facetable (all fine on Image)
  - [ ] Ranger District - with Document type, missing from showpage, not searchable, not facetable (all fine on Image)
  - [ ] TGN - with Document type, missing from showpage, not searchable, not facetable (all fine on Image)
  - [ ] Water Basin - with Document type, missing from showpage, not searchable, not facetable (all fine on Image)
- [x] Dates - pass
- [ ] Identifiers - fail
  - [ ] Hydrologic Unit Code (HUC) - missing from showpage
  - [ ] Longitude Latitude Identification (LLID) - missing from showpage
- [x] Rights - pass
- [ ] Source (Publishing) - fail
  - [ ] Citation - missing from showpage
  - [ ] Language - with Document type, missing from showpage, not searchable, not facetable (all fine on Image)
  - [ ] Local Collection Name - with Document type, missing from showpage, not searchable, not facetable (all fine on Image)
  - [ ] Publisher - with Document type, missing from showpage, not searchable, not facetable (all fine on Image)
  - [ ] Place of Production - form has text field, not URI field
  - [ ] Publication Place - form has text field, not URI field
  - [ ] Repository - form value not saving (Brandon looking into this one, may be addressing this in another PR)
- [ ] Relations - fail
  - [ ] Art Series - missing from showpage
  - [ ] Has Part - missing from showpage
  - [ ] Is Version Of - missing from showpage
  - [ ] Larger Work - missing from edit form
- [x] Types - pass
- [x] Formats - pass
- [x] Groupings - pass
- [ ] Administratives - fail
  - [ ] Color Content - missing from showpage
  - [ ] Color Space - missing from showpage
  - [ ] Conversion - missing from showpage
  - [ ] Date Digitized - missing from showpage
  - [ ] File Size - missing from showpage
  - [ ] Height - missing from showpage
  - [ ] Modified - missing from showpage
  - [ ] Original Filename - missing from edit form
  - [ ] Resolution - missing from showpage
  - [ ] Submission Date - missing from edit form
  - [ ] Width - missing from showpage

**Comment by raybrarian** [comment]
New QA list - each one should be able to be:

- searched
- display on the showpage (check Document type and Image type)
- show on the edit form
- show as a facet

- [ ] Titles 
- [ ] Creators 
- [ ] Descriptions 
  - [ ] Abstract
  - [ ] Compass Direction
  - [ ] Description
  - [ ] Form of Work
  - [ ] Identification Verification Status
  - [ ] Layout
  - [ ] Mode of Issuance
  - [ ] MODS Note
  - [ ] Number of Pages
  - [ ] Object Orientation
  - [ ] Tribal Notes
  - [ ] Source Condition
  - [ ] Specimen Type
  - [ ] Temporal
- [ ] Subjects
- [ ] Geographics
  - [ ] Box
  - [ ] GPS Latitude
  - [ ] GPS Longitude
  - [ ] Location - with Document type
  - [ ] Ranger District - with Document type
  - [ ] TGN - with Document type
  - [ ] Water Basin - with Document type
- [ ] Dates
- [ ] Identifiers
  - [ ] Hydrologic Unit Code (HUC)
  - [ ] Longitude Latitude Identification (LLID)
- [ ] Rights
- [ ] Source (Publishing)
  - [ ] Citation
  - [ ] Language - with Document type
  - [ ] Local Collection Name
  - [ ] Publisher
  - [ ] Place of Production
  - [ ] Publication Place
  - [ ] Repository
- [ ] Relations
  - [ ] Art Series
  - [ ] Has Part
  - [ ] Is Version Of
  - [ ] Larger Work
- [ ] Types
- [ ] Formats
- [ ] Groupings
- [ ] Administratives
  - [ ] Color Content
  - [ ] Color Space
  - [ ] Conversion
  - [ ] Date Digitized
  - [ ] File Size
  - [ ] Height
  - [ ] Modified
  - [ ] Original Filename
  - [ ] Resolution
  - [ ] Submission Date
  - [ ] Width

**Comment by raybrarian** [comment]
Metadata will QA.

**Comment by sseymore** [comment]
Please see QA passes above. 

Notes on failures:

Descriptions:
Specimen Type: Missing help text

Subjects:
State/Edition: Not on show page, but is on edit form.

Geographics:
Location: Not in search results
TGN: Showpage has field label as "Geographic Name"

Dates: All pass except Collection Date: pass, but property label in form and show page display is "Collection Date” Please change it back to Collected Date or Date Collected.

Rights:
Use Restrictions: Not on show page, but is on edit form.
Access Restrictions: Not on show page, but is on edit form.

Source:
Language: Not searchable.

Relations: All pass except Relation: pass, but label in form and display is "Relations"

Types:
Type: Not searchable

Formats:
Extent: not on show page, but is on edit form
Measurements: not on show page, but is on edit form
Physical Extent: not on show page, but is on edit form

Administrative:
File Size: not in form or on show page
Submission Date: not in form or on show page
Contents: not on show page, but is on edit form

**Comment by raybrarian** [comment]
@CGillen @straleyb - I don't know if either one of you can take a look at this today (and I'm not expecting that!), but I wanted to flag Sarah's notes above about pass/fail. Unless I hear otherwise, I'm going to move this to WC7 at the close of this WC.

**Comment by shieldsb** [comment]
@straleyb @CGillen Metadata did QA on this at the very end of WC 6. There a few items above that need review, so we've it back to the Ready column. This is related to #397 that was also moved back to Ready.

**Comment by raybrarian** [comment]
@sseymore - This should be ready for the remaining QA (8/23 failures list should be fixed). If someone in Metadata could take a look? Thanks!

**Comment by sseymore** [comment]
Note: waiting for seed data ingest to QA

**Comment by sseymore** [comment]
@raybrarian @shieldsb I can't QA all of these fields now because I can't add a new work (file upload is failing) with some of the data that failed above to search on. I will check back in on this at the beginning of WC9.

**Comment by shieldsb** [comment]
@sseymore @raybrarian I can add new works and the files upload, but the thumbnails don't display in the search results and the image does not display during the submission process.  I can see the work on the show page once it's deposited and approved.  It's been this way for nearly a week.  I'm going to create a new ticket and add it WC9.  Are you having similar issues, or will it not allow up you to upload/deposit new works at all?

**Comment by shieldsb** [comment]
@sseymore The file upload issue has been resolved.

**Comment by sseymore** [comment]
@shieldsb thanks, I can QA this next week.

**Comment by sseymore** [comment]
Notes on failures:

Subjects:
State/Edition: Not is search results or show page, but is on edit form. When adding the item and inputting metadata in the form, the data does not make it to live object. 

Geographics:
Location: Not in search results or show page. 

Dates: 
Collected Date: Not in search results or show page, but is on edit form. When adding the item and inputting metadata in the form, the data does not make it to live object or stay with item. 

Formats:
Extent: On show page, but not searchable.
Measurements: On show page, but not searchable.
Physical Extent: On show page, but not searchable.

Administrative:
File Size: On show page, but not searchable. This might move to File Set metadata only. 
Submission Date: Not on show page, not searchable. This might move to File Set metadata only. 
Contents: Not on show page/searchable, but is on edit form. This is in the RDF for cpds, so N/A.

**Comment by sseymore** [comment]
The metadata team also identified another error on the show page: Material, Measurements, Technique, Extent, and Physical Extent are appearing twice.

**Comment by CGillen** [comment]
Fixed by #961
Subjects:
- [x] State/Edition: Already exists on show page. Not requested to be on search results page.

Geographics:
- [x] Location: Fixed by #943 

Dates:
- [x] Collected Date: Added to show page

Formats:
- [x] Extent: Already exists on show page. Requested to not be searchable.
- [x] Measurements: Already exists on show page. Requested to not be searchable.
- [x] Physical Extent: Already exists on show page. Requested to not be searchable.

Administrative:
- [x] File Size: Already exists on show page. Requested to not be searchable.
- [x] Submission Date: Had submission date on the form, Hyrax automatically sets if not on the form. Removed from form, newly submitted works will have Submission Date display on show page.
- [x] Contents: Requested to not be on show page. Requested to not be searchable.

Duplicates:
- [x] Material
- [x] Measurements
- [x] Technique
- [x] Extent
- [x] Physical Extent

**Comment by sseymore** [comment]
QA pass!! Apologies for not confirming with the MAP on the searchable-ness. I think repetitive QA tripped me up.

**Comment by raybrarian** [comment]
@sseymore - you ROCK!
